noxfile.py

```diff
@@ -0,0 +1,77 @@
+import os
+from pathlib import Path
+
+import nox
+
+LINT_FILES = ("hacking/pr_labeler/label.py", "noxfile.py")
+PINNED = os.environ.get("PINNED", "true").lower() in {"1", "true"}
```

Comment:
@gotmax23 when you get a min, can you explain what this is doing? I see what's going on with this line of code but when is "PINNED" set as an environment variable?

Reply:
If a user wants to test without dependency pins, they can set `PINNED=0`.

Reply:
Thanks for the explanation. We can do it in a follow up PR but it'd be ideal to add some user doc on points like that in the README.

Reply:
@oraNod that'd look like `PINNED=0 nox -s some-thing`.

FWIW I like how pip has hints for different workflow process invocations, like https://github.com/pypa/pip/blob/d8cd93f/noxfile.py#L269-L276.

Reply:
@webknjaz cheers. it suddenly felt kind of obvs after @gotmax23's explanation but maybe I just didn't consider that pattern of interaction. I like those hints too.